sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/PushDownUtils.scala

```diff
@@ -117,7 +145,11 @@ object PushDownUtils extends PredicateHelper {
       relation: DataSourceV2Relation,
       projects: Seq[NamedExpression],
       filters: Seq[Expression]): (Scan, Seq[AttributeReference]) = {
-    val exprs = projects ++ filters
+    val filePartitionFilter = scanBuilder match {
+      case fileScanBuilder: FileScanBuilder => fileScanBuilder.getPartitionFilters
```

Comment:
```
If we don't want to touch PruneFileSourcePartitions, I guess instead of checking if the postScanFilter is empty using if filters.isEmpty, we do something like this:

if (JDBC)
  if filters.isEmpty
     push down aggregate
else // file based
  if filters are only partition filters
    push down aggregate
This looks hacky though. Please let me know if anybody has a better idea.
```
@huaxingao I don't have a better idea either. 
But you have to match special case `FileScanBuilder` here anyway. I would prefer resolving the partition filters in `pushAggregates`, which is more straightforward.

Reply:
@gengliangwang OK. I will not change `PruneFileSourcePartitions`. I will close this PR and find other way to do this. Thank you very much for your help!